### PR TITLE
Extends :full, :half, :quarter to l,t,r,b

### PR DESCRIPTION
**Pull request by squidpunch**
this extends the functionality in #185 to include the other positioning elements so you can do things like `st.frame = { t: :half }`  etc

### CODE REVIEW

**Comment by squidpunch** [comment]
So I am not 100% sure the best way to benchmark this stuff, but here is what I did:
* use `rm-bencharking` (https://github.com/squidpunch/rm-benchmarking) which i think was something I forked from @GantMan but I dont even recall at this point (I thanked him, so at least the idea came from something of his :smile: )
* ran the following script in that benchmark against a local copy of the branch created by @markrickert to get a baseline
```ruby
      a = UIView.alloc.initWithFrame([[0,0],[rmq.device.width, rmq.device.height]])
      b = UIView.alloc.init
      rmq(a).append(b)

      40.times do
        Benchmark.bm do |x|
          x.report { 20000.times { rmq(b).style { |st| st.frame = { w: :full } } } }
        end
      end
```
* This gives me 40 instances of running the style method 20000 times, so we can check min, max, and averages
* Changed to my branch with my changes and ran the same script
* In both cases I actually ran it twice, just so that if there was any sort of caching or anything like that, they would both benefit from the same.

Results are as follows:

All times are "real" times from the benchmarking

```
My Avg: 0.1676862
Mark Avg: 0.151477243902439
	
My Max: 0.260607
Mark Max	: 0.178996
	
My Min: 0.144440
Mark Min: 0.136839
```

so looking at the numbers we did add some extra time to the processing with this change.  but on average they are very close in general.  I could take a pass at instead of defining the method as I have, to make it less DRY and repeat the same thing in the main method, and see if that changes the numbers - or perhaps this is such a small change in average that it covers what was asked for - without impacting performance greatly enough to be a concern?

**Comment by GantMan** [comment]
:)

**Comment by twerth** [comment]
I'll review this tomorrow :-)

**Comment by squidpunch** [comment]
thanks @twerth let me know what you think and anything that we may need to adjust!

File: motion/ruby_motion_query/rect.rb [hunk 1]
```diff
@@ -202,35 +202,28 @@ def rect_hash_to_rect_array(view, existing_rect, params, grid = nil)
...
-          else
-            puts "\n[RMQ ERROR] height symbol #{params_h} can't be parsed.\n\n"
-          end
-        end
+        params_l = value_from_symbol(params_l, sv.frame.size.width) if params_l.is_a?(Symbol) && sv
+        params_t = value_from_symbol(params_t, sv.frame.size.height) if params_t.is_a?(Symbol) && sv
+        params_w = value_from_symbol(params_w, sv.frame.size.width) if params_w.is_a?(Symbol) && sv
+        params_h = value_from_symbol(params_h, sv.frame.size.height) if params_h.is_a?(Symbol) && sv
+        params_r = value_from_symbol(params_r, sv.frame.size.width) if params_r.is_a?(Symbol) && sv
+        params_b = value_from_symbol(params_b, sv.frame.size.height) if params_b.is_a?(Symbol) && sv
```
Comment (twerth) [inline motion/ruby_motion_query/rect.rb]: ^ none of this needs to be called if there is no sv (213 through 218)

Comment (squidpunch) [inline motion/ruby_motion_query/rect.rb]: Good call. I'll update this evening.

File: motion/ruby_motion_query/rect.rb [hunk 2]
```diff
@@ -376,6 +369,15 @@ def rect_hash_to_rect_array(view, existing_rect, params, grid = nil)
         [l,t,w,h]
       end
 
+      def value_from_symbol(size_symbol, max_value)
+        width_height_symbols = {full: 1, half: 0.5, quarter: 0.25}
```
Comment (twerth) [inline motion/ruby_motion_query/rect.rb]: This is nice code of course, but I'm guessing an if/elsif/elsif would be faster than creating a hash each time; Performance-wise. Every ms we can squeeze out of this helps on every view in every app :-)

Comment (squidpunch) [inline motion/ruby_motion_query/rect.rb]: I'll change and benchmark it! Thanks Todd.

**Comment by squidpunch** [comment]
I have updated with your feedback @twerth new benchmarks 

image

closer to what we started with in the changes from Mark than before for sure, with a small increase in average and max.

**Pull request merged**